src/pages/namespaces.html

```diff
@@ -120,7 +120,8 @@ <h1>
                     <button class="btn btn-link edit-namespace-btn" ng-click="rowform.$show()" tooltip-trigger="mouseenter"
                             tooltip="{{'edit.namespace' | translate}}"><span class="icon-edit"></span></button>
                     <button class="btn btn-link secondary delete-namespace-btn" ng-click="removeNamespace(namespace)"
-                            ng-disabled="!canWriteActiveRepo()" tooltip-trigger="mouseenter" tooltip="{{'delete.namespace' | translate}}">
+                            ng-disabled="{{!canWriteActiveRepo()}}" tooltip-trigger="mouseenter" tooltip="{{'delete.namespace' | translate}}"
+                            data-cy="{{addCustomId(namespace.prefix)}}">
```

Comment:
You don't need these `addCustomId` methods in the controller. You can just do it inline like
```data-cy="{{'delete-pred_' + namespeace.prefix}}"```

Also bare in mind that it's really bad practice to bind to functions in the controller, because those functions are executed quite a lot with each digest cycle which might really slow down the rendering of the page. So, in short, you must avoid this.